aio.api.github/aio/api/github/abstract/commit.py

```diff
@@ -0,0 +1,18 @@
+
+from datetime import datetime
+from functools import cached_property
+
+import abstracts
+
+from aio.api.github import utils
+from .base import GithubRepoEntity
+
+
+class AGithubCommit(GithubRepoEntity, metaclass=abstracts.Abstraction):
+    """A Github commit."""
+
+    @cached_property
+    def timestamp(self) -> datetime:
+        """Datetime of the commit."""
+        return utils.dt_from_js_isoformat(
+            self.data["commit"]["committer"]["date"])
```

Comment:
will there be a chance for KeyError that needs to be guarded?

Reply:
i guess so, altho not if this is called with the correct data

it feels like this should have some error handling, but tbh not sure what it would raise exactly - eg in the normal case a the `data` dict is -> `obj.attrs` - so should it raise an `AttributeError` as would be the normal case of getting an attr from the data

perhaps we need to make it so that if a key is missing it raises a more specific error like `GithubDataError` or somesuch

Reply:
i guess the reason i didnt do this already is that we dont want these errors to be handled - at least not as upstream-type errors

ie if this `data` doesnt have the expected keys its unlikely to be because the github api has started returning incorrect data and more likely a programming error so we would want to see the stack trace